### ISSUE TITLE
pxeboot/fbnetboot:could not run

### ISSUE THREAD

**Issue by hanxinwu1** [u-root/u-root#2511]
**Describe the bug**
when I use pxeboot or fbnetboot like this
![image](https://user-images.githubusercontent.com/102037445/194033471-58dc6518-e190-44a0-b448-5bf408ebe1b7.png)
The command-line output is “could not configure eth0 for ipv4,unable to open a broadcasting socket: address family not supported by protocol” 
My network card is Intel's and the driver is loaded, do you know what causes it, is this a bug?
And
We all know that there is ipconfig to configure ip in Linux systems, in the initramfs made by u-root, what is used to configure the IP address? Is there only dhclient?
![47b54783f6e4ddff60852e083ccc500](https://user-images.githubusercontent.com/102037445/194102898-e6a58258-d799-4542-8336-81a83bcea287.jpg)

Thanks!

**Comment by hanxinwu1** [comment]
Could not configure eth0 for IPv4: unable to open a broadcasting socket: address family not supported by protocol.
Is linux kernel's net part causing this problem?And how can I solve it？Can you think of any workarounds？Thank you!

**Comment by hanxinwu1** [comment]
And when I use /cmd/boot/fbnetboot,that mean use fbnetboot,I get this log.
![image](https://user-images.githubusercontent.com/102037445/194590432-d89e552e-1f80-4c89-83a9-629935544f68.png)
Is this the reason I didn't set up a pxe server? How to set up a good fbnetboot server, the same as pxeboot? What is the difference between pxeboot and fbnetboot?

**Comment by Keno** [comment]
Check whether your kernel config has `CONFIG_PACKET` set. I had the same issue and setting that fixed it.

**Comment by hanxinwu1** [comment]
> 

Yes, it works. Another question I would like to ask is that when I use dhclient, sometimes I can correctly assign an ipv4 address but sometimes I will report the following errors.
![cba27d7279bd5ce0d9e8bd824a90ef0](https://user-images.githubusercontent.com/102037445/195110464-49d65687-1ca1-4cdd-837b-76afdd5c1bef.jpg)
Do you know why?

**Comment by Keno** [comment]
Yes, you're missing entropy. Assuming you're in qemu, you should configure your kernel to support the virtio entropy source and then add that to the qemu launch line.

**Comment by hanxinwu1** [comment]
> Yes, you're missing entropy. Assuming you're in qemu, you should configure your kernel to support the virtio entropy source and then add that to the qemu launch line.

I am running in the server results. When I type this command, sometimes it works, sometimes it doesn't work.
Then when it runs successfully, enters the new kernel interface, and starts to download the system, it has an error.
Error :probing for devices to install to failed (sata disks cannot be identified)
![mmexport1665506850374](https://user-images.githubusercontent.com/102037445/195151818-37fa311a-fdac-4f28-9870-36d551fa55d9.jpg)

Do you know that's why?(I guess :Is there no sata hard disk drive for the ubuntu operating system ubuntu22.10 downloaded from my website)

**Comment by hanxinwu1** [comment]
> 

I tried the pxeboot command to install the system today for another day, and I ran it on a physical server. I tried ubuntu 22.10 again, and I was already on the interface for configuring the language; 
![6d3014c05345b9fd64d82f94b956dda](https://user-images.githubusercontent.com/102037445/195351113-003124a9-f8bb-444b-b218-c21cfcd56b17.jpg)

but the error after that was: probing for devices to install to failed.
![e52b799dc613e408eac643ddc68e42e](https://user-images.githubusercontent.com/102037445/195351492-90233f50-f734-4c69-adf6-1c7c199cd816.jpg)

 I also tried to install centos7 or centos8, which had an error before the kernel initialization had started http downloading the iso file, both of which were reported as: no carrier detected on interface eth0.
![7d9aaec5d8e48fcdfb611baa8e9433a](https://user-images.githubusercontent.com/102037445/195351739-eea213b3-08f2-4b36-a878-a1487b2e0cc3.jpg)

 I hadn't done a pxe installation system before that, which was a bit tricky for me.It's a more annoying thing.Keep up the good work tomorrow haha.
